### PR TITLE
#9386: Include the rustfmt.toml when generating the node-template

### LINKED ISSUE #9386: Include the `rustfmt.toml` when generating the `node-template`
With #8982 and #9381, the `node-template` now will be formatted with the same rustfmt settings, and we would like to propagate this file in the [the template generation tool](https://github.com/paritytech/substrate/blob/master/.maintain/node-template-release/src/main.rs) so that the [devhub node template](https://github.com/substrate-developer-hub/substrate-node-template) will always include the same `rustfmt.toml` file.

### PR DESCRIPTION
**Pull request by ericjmiller**
Fixes #9386 
**Description**
The [node template generator](https://github.com/paritytech/substrate/blob/master/.maintain/node-template-release/src/main.rs) now adds the Substrate `rustfmt.toml` at the root level.

Oh, and I couldn't resist running `rustfmt` on this file.

**Important**
I used relative pathing to locate the Substrate `rustfmt.toml`.  Otherwise, it's simple and straightforward.

### CODE REVIEW

File: .maintain/node-template-release/src/main.rs [hunk 1]
```diff
@@ -233,20 +250,28 @@ fn main() {
...
 
 			update_top_level_cargo_toml(&mut cargo_toml, workspace_members, &node_template_path);
 		}
 
 		write_cargo_toml(&t, cargo_toml);
 	});
 
+	// ensures rustfmt is consistent across repositories.
+    	let node_template_rustfmt_toml_path = node_template_path.join("rustfmt.toml");
+    	let rustfmt_target = Path::new("../../rustfmt.toml");
```
Comment (bkchr) [inline .maintain/node-template-release/src/main.rs:264]: This needs to be relative to `options.node_template`

**Review by bkchr — CHANGES_REQUESTED** [review]

File: .maintain/node-template-release/src/main.rs (hunk 1 above)
Comment (ericjmiller) [reply .maintain/node-template-release/src/main.rs:264]: Thanks, @bkchr.  Perhaps I misunderstand the issue here.

I thought the `options.node_template` didn't have the canonical `rustfmt.toml` and therefore required the Substrate repo's `rustfmt.toml`.  In which case, there isn't a guaranteed way to get to the `substrate/rustfmt.toml` from the `options.node_template` path.

Comment (ericjmiller) [reply .maintain/node-template-release/src/main.rs:264]: Okay, I believe I understand my confusion.  When `node-template` is referenced in this issue, it's speaking of `bin/node-template` and therefore exists in this repo.  I mistakenly thought it was the [devhub node-tempate](https://github.com/substrate-developer-hub/substrate-node-template) from another (related) repo passed in as an argument.

Running some QA on the fix and will commit soon.

**Comment by bkchr** [comment]
@ericjmiller can you please merge master again. You seem to have fucked up something :grimacing:

**Comment by ericjmiller** [comment]
No idea what's going on.  My local repo looks fine, then I push to remote and see the large quantity of changes in the PR.  very confusing.  I may have to just destroy this and recreate.

**Pull request closed without merging**